homeassistant/components/automation/device_automation.py

```diff
@@ -0,0 +1,73 @@
+"""Helpers for device automations."""
+import importlib
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components import websocket_api
+from homeassistant.core import split_entity_id
+from homeassistant.helpers.entity_registry import async_entries_for_device
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def _is_domain(entity, domain):
+    return split_entity_id(entity.entity_id)[0] == domain
+
+
+async def async_setup(hass):
+    """Set up device automation."""
+    hass.components.websocket_api.async_register_command(
+        websocket_device_automation_list_triggers)
+    return True
+
+
+async def async_get_device_automation_triggers(hass, device_id):
+    """List device triggers."""
+    device_registry = await hass.helpers.device_registry.async_get_registry()
+    entity_registry = await hass.helpers.entity_registry.async_get_registry()
+
+    domains = set()
+    triggers = []
+    device = device_registry.async_get(device_id)
+    for entry_id in device.config_entries:
+        config_entry = hass.config_entries.async_get_entry(entry_id)
+        domains.add(config_entry.domain)
+
+    entities = async_entries_for_device(entity_registry, device_id)
+    for entity in entities:
+        domains.add(split_entity_id(entity.entity_id)[0])
+
+    for domain in domains:
+        module = None
+        try:
+            module = importlib.import_module(
+                '...{}.device_automation'.format(domain), __name__)
+        except ImportError:
+            _LOGGER.exception("Invalid domain %s", '...{}'.format(domain))
+            continue
+
+        if hasattr(module, 'DOMAIN_TRIGGERS'):
+            pass
+        if hasattr(module, 'ENTITY_TRIGGERS'):
```

Comment:
We shouldn't do this, as different entities have different supported features. Instead, we should call a function:

```python
platform = integration.get_platform('device_automation')

triggers.extend(platform.async_get_triggers(hass, device_id)
```

That way, all logic lives inside integrations, and we don't have to define any "structure" besides the result structure.